barretenberg/cpp/src/barretenberg/dsl/acir_format/ivc_recursion_constraint.cpp

```diff
@@ -290,7 +290,7 @@ void populate_dummy_vk_in_constraint(MegaCircuitBuilder& builder,
 
     // Add the fields to the witness and set the key witness indices accordingly
     for (auto [witness_idx, value] : zip_view(key_witness_indices, mock_vk_fields)) {
-        witness_idx = builder.add_variable(value);
+        builder.assert_equal(builder.add_variable(value), witness_idx);
```

Comment:
This is the other core change. The problem here was the VK witnesses are used in multiple place (i.e. business logic + recursive verification) and those uses are connected via copy constraints. Directly overwriting the witness indices was quietly disconnecting those copy constraints. Using `assert_equal` maintains them

Reply:
I wonder whether there is a way to add a test to enforce this.

Reply:
yes, in fact trying to add such a test was how I found these two issues in the first place. Problem is there are more of them. Once I fix them all then the test can go in